### PR TITLE
Prod-2083 Review & Payment -> Develop

### PR DESCRIPTION
**Pull request by mmattlin**
## What's in This PR
- The purpose of this PR was to re-merge the changes for PROD-2083 - Implement the Review and Payment page for each of the work intakes to work and look like the mockups - that had been overwritten.
- However, notice that it is also contains changes to add the new Web Design and Data Advisory Intakes
- I also cleaned up the BasicInfoForm file that unnecessary conditionals to differentiate between work intakes

### CODE REVIEW

File: src/constants/index.js [hunk 1]
```diff
@@ -355,33 +483,33 @@ export const webWorkTypes = [
...
-    title: "Website Development",
-    subTitle: "Information to help understand what this category would include",
-    price: 499,
-  },
-  {
-    title: "Web App Design",
-    subTitle: "Information to help understand what this category would include",
-    price: 499,
-  },
+  // {
```
Comment (brooketopcoder) [inline src/constants/index.js]: I generally don't like to leave commented code in the app. Can we either delete this entirely or add a TODO: w/specifics for when we can uncomment it?

File: src/hooks/useCheckMobileScreen.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  };
+
+  useEffect(() => {
+    window.addEventListener("resize", handleWindowSizeChange);
+    return () => {
+      window.removeEventListener("resize", handleWindowSizeChange);
+    };
+  }, []);
+
+  return width <= 768;
```
Comment (brooketopcoder) [inline src/hooks/useCheckMobileScreen.js:18]: This is a not a breakpoint we currently support. This also seems like a lot of extra complexity just to resize a modal. 

i don't think we need to fix any of this now, but can you add a TODO: to either deprecate and remove this hook or to fix the breakpoint so it's one we support. 

I don't want people coming along and re-using this hook or potentially refactoring into the /src-ts dir.

File: src/routes/Branding/components/ColorOptions/index.jsx [hunk 3]
```diff
@@ -30,9 +31,19 @@ const ColorOptions = ({ colors, selectedColor, onSelect }) => {
               );
               onSelect(newColors, newColors);
             } else {
-              if (selectedColor.value.length >= 3) return;
-              const newColors = [...selectedColor.value, color.name];
-              onSelect(newColors, newColors);
+              if (color.isAny) {
```
Comment (brooketopcoder) [inline src/routes/Branding/components/ColorOptions/index.jsx]: I know you didn't add this, but it seems like unnecessary nesting. Would the following work?


```
 onClick={() => {
            if (!_.isArray(selectedColor.value)) {
              selectedColor.value = [];
              selectedColor.option = [];
            }
            if (_.includes(selectedColor.value, color.name)) {
              const newColors = _.filter(
                selectedColor.value,
                (v) => v !== color.name
              );
              onSelect(newColors, newColors);
            } else if (color.isAny) {
              const newColors = [color.name];
              onSelect(newColors, newColors);
            } else if (selectedColor.value.length < 3) {
              const newColors = [
                ...selectedColor.value.filter(
                  (name) => name !== anyColor?.name
                ),
                color.name,
              ];
              onSelect(newColors, newColors);
            }
          }}

File: src/routes/Payment/components/PaymentForm/index.jsx [hunk 4]
```diff
@@ -8,9 +8,9 @@ import PT from "prop-types";
 import React, { useState } from "react";
 import { useSelector } from "react-redux";
 
-import { FormField } from "../../../../components/FormElements/FormField";
-import { FormInputText } from "../../../../components/FormElements/FormInputText";
-import { ReactSelect } from "../../../../components/ReactSelect";
+import FormField from "components/FormElements/FormField";
```
Comment (brooketopcoder) [inline src/routes/Payment/components/PaymentForm/index.jsx]: Again, I know you didn't make this change, but I prefer the relative import paths bc visual studio code can't navigate to a definition of absolute paths.

File: src/routes/Products/components/BasicInfo/index.jsx [hunk 5]
```diff
@@ -46,27 +53,51 @@ const BasicInfo = ({
...
+        website: { title: "Website Address", value: "", option: "" },
+        feedback: { title: "What Do You Like", value: "", option: "" },
+      },
+    ],
     sampleData: { title: "Sample Data", option: "", value: "" },
-  });
+  };
+
+  const [formData, setFormData] = useState(defaultFormData);
+  const isFindMeData = bannerData.title === "Find Me Data";
```
Comment (brooketopcoder) [inline src/routes/Products/components/BasicInfo/index.jsx]: can we use the WorkType enum here?

File: src/routes/Review/components/ReviewTable/index.jsx [hunk 6]
```diff
@@ -1,174 +1,39 @@
-import { Link } from "@reach/router";
-import classNames from "classnames";
-import PageDivider from "components/PageDivider";
-import { ProgressLevels } from "constants/";
-import React, { useState } from "react";
-import _ from "lodash";
-import ArrowIcon from "../../../../assets/images/icon-arrow.svg";
-import "./styles.module.scss";
+import React from "react";
+import { WorkType } from "../../../../../src-ts/lib";
```
Comment (brooketopcoder) [inline src/routes/Review/components/ReviewTable/index.jsx]: let's import this from the /src-ts directory and not the /src-ts/lib

File: src/routes/Review/components/ReviewTable/index.jsx [hunk 7]
```diff
@@ -1,174 +1,39 @@
...
+import { WorkDetailDetailsPane } from "../../../../../src-ts/tools/work/work-detail-details/work-detail-details-pane";
```
Comment (brooketopcoder) [inline src/routes/Review/components/ReviewTable/index.jsx]: We should not be importing anything from the this deep into the tools. Let's add this to the import from /src-ts

File: src/routes/Review/index.jsx [hunk 8]
```diff
@@ -75,6 +75,8 @@ const Review = ({
   });
 
   const isDataExploration = bannerData.title === "Data Exploration";
+  const isDataAdvisory =
```
Comment (brooketopcoder) [inline src/routes/Review/index.jsx:78]: let's use the WorkType enum here

File: src/routes/Review/index.jsx [hunk 9]
```diff
@@ -146,10 +150,10 @@ const Review = ({
       "form.basicInfo.selectedDevice.option.length",
       1
     );
-    const additionalPaymentInfo =
-      workType?.selectedWorkType === "Website Design"
-        ? `\n${numOfPages} Pages\n${numOfDevices} Devices`
-        : "";
+    const additionalPaymentInfo = "";
+    // workType?.selectedWorkType === "Website Design"
```
Comment (brooketopcoder) [inline src/routes/Review/index.jsx]: can we delete this comment out code or add a TODO: to explicitly describe when we can uncomment it?

File: src/routes/SelectWorkType/index.jsx [hunk 10]
```diff
@@ -6,28 +6,116 @@ import { connect, useDispatch, useSelector } from "react-redux";
 import { triggerAutoSave } from "../../actions/autoSave";
 import { saveWorkType, toggleSupportModal } from "../../actions/form";
 import { setProgressItem } from "../../actions/progress";
-import BackIcon from "../../assets/images/icon-back-arrow.svg";
-import IconWebsiteTools from "../../assets/images/design-tools.svg";
+// import BackIcon from "../../assets/images/icon-back-arrow.svg";
```
Comment (brooketopcoder) [inline src/routes/SelectWorkType/index.jsx]: ditto about all the comments in this file

**Review by brooketopcoder — CHANGES_REQUESTED** [review]
This looks fantastic--thanks for figuring it all out.

The Social SVGs aren't used anywhere. I am guessing these were added by Lunarkid bc he wasn't aware that we'd already built the footer.

The website design icon also already exists in the /src-ts/lib/svgs.

The rest of the suggestions are super minor.

**Review by testflyjets — APPROVED** [review]

File: src/routes/Review/index.jsx (hunk 9 above)
Comment (mmattlin) [reply src/routes/Review/index.jsx]: Resolved

File: src/routes/Review/index.jsx (hunk 8 above)
Comment (mmattlin) [reply src/routes/Review/index.jsx:78]: Resolved

File: src/routes/Review/components/ReviewTable/index.jsx (hunk 7 above)
Comment (mmattlin) [reply src/routes/Review/components/ReviewTable/index.jsx]: Done

File: src/routes/Payment/components/PaymentForm/index.jsx (hunk 4 above)
Comment (mmattlin) [reply src/routes/Payment/components/PaymentForm/index.jsx]: Resolved

File: src/routes/Branding/components/ColorOptions/index.jsx (hunk 3 above)
Comment (mmattlin) [reply src/routes/Branding/components/ColorOptions/index.jsx]: Resolved

File: src/hooks/useCheckMobileScreen.js (hunk 2 above)
Comment (mmattlin) [reply src/hooks/useCheckMobileScreen.js:18]: Resolved

**Review by brooketopcoder — APPROVED** [review]
Perfecto! I just removed the obsolete website design icon, and we're good to go. Thanks!

**Pull request merged**